### PR TITLE
Add Mock GitHub API for testing purposes

### PR DESCRIPTION
**Pull request by noorbuchi**
This pull requests helps in avoiding the use of the actual GitHub API when running unit tests. It adds the assumption that the REST API provided by GitHub works as expected. Additionally, it helps in the development of this tool in the following ways:
1. Tests can run much faster
2. Tests are unlikely to have intermittent failures caused by multiple sessions running at the same time
3. high test coverage is maintained
4. No need to create mandatory teardown at test failures
This is an early stage of the mock API, while it currently functions as expected, it will likely need to be expanded as new functionalities are added.

Closes: #27  #26  #23

### CODE REVIEW

**Review by antlet — APPROVED** [review]
This all looks great! Ready to merge on approval.

**Comment by noorbuchi** [comment]
I just discovered an Issue with this pull request where the mock API was not utilized in all the places that it should have. I will be working on adding the fix before we merge this. With that in mind, I will ask again for reviews once the new changes are ready.

**Comment by noorbuchi** [comment]
I just fixed the previously mentioned issue. This PR is ready to merge once it gets all the needed approvals.

**Pull request merged**
